### PR TITLE
fix: username password flow

### PR DESCRIPTION
**Pull request by mohanjith**
Adds the feature requested in https://github.com/AzureAD/microsoft-authentication-library-for-go/issues/370

### CODE REVIEW

**Comment by mohanjith** [comment]
@microsoft-github-policy-service agree

**Comment by mohanjith** [comment]
@microsoft-github-policy-service agree

**Comment by 4gust** [comment]
The changes look good overall; however, I noticed that there aren't any tests included for the new function. Could you please add some test cases to ensure the function works as expected and to help prevent regressions in the future?

Having tests will also make it easier to verify the changes and ensure compatibility with the existing codebase. Let me know if you need any guidance on setting up the tests or the testing framework.

File: apps/confidential/confidential.go [hunk 1]
```diff
@@ -592,6 +600,46 @@ func (cca Client) AcquireTokenSilent(ctx context.Context, scopes []string, opts
...
 }
 
+// acquireTokenByUsernamePasswordOptions contains optional configuration for AcquireTokenByUsernamePassword
+type acquireTokenByUsernamePasswordOptions struct {
+	claims, tenantID string
+	authnScheme      AuthenticationScheme
+}
+
+// AcquireByUsernamePasswordOption is implemented by options for AcquireTokenByUsernamePassword
+type AcquireByUsernamePasswordOption interface {
```
Comment (bgavrilMS) [inline apps/confidential/confidential.go:610]: We would need some unit tests where the HTTP is mocked. The tests should cover use of `AcquireTokenByUsernamePassword` and `AcquireTokenSilent`

Comment (mohanjith) [reply apps/confidential/confidential.go:610]: I haven't changed AcquireTokenSilent. I added tests for AcquireTokenByUsernamePassword

**Comment by mohanjith** [comment]
@4gust I have added similar tests as in public client. Let me know if you would like to to write more tests.

Thank you!

**Review by 4gust — APPROVED** [review]
Looks good to me!

**Review by bgavrilMS — APPROVED** [review]

**Pull request merged**